restorer.go

```diff
@@ -117,3 +159,68 @@ func (r *Restorer) restoreLayer(cache Cache, sha string) error {
 
 	return layers.Extract(rc, "")
 }
+
+func (r *Restorer) restoreSBOM() error {
+	var (
+		cacheDir  = filepath.Join(r.LayersDir, "sbom", "cache")
+		launchDir = filepath.Join(r.LayersDir, "sbom", "launch")
+	)
+
+	defer os.RemoveAll(filepath.Join(r.LayersDir, "sbom"))
+
+	err := filepath.Walk(cacheDir, r.restoreSBOMFunc("cache"))
+	if err != nil {
+		return err
+	}
+
+	err = filepath.Walk(launchDir, r.restoreSBOMFunc("launch"))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Restorer) restoreSBOMFunc(bomType string) func(path string, info fs.FileInfo, err error) error {
+	var bomRegex *regexp.Regexp
+
+	if runtime.GOOS == "windows" {
+		bomRegex = regexp.MustCompile(fmt.Sprintf(`%s\\(.+)\\(.+)\\(bom.+json)`, bomType))
+	} else {
+		bomRegex = regexp.MustCompile(fmt.Sprintf(`%s/(.+)/(.+)/(bom.+json)`, bomType))
+	}
+
+	return func(path string, info fs.FileInfo, err error) error {
+		if info != nil && info.IsDir() {
```

Comment:
Should we verify this is a file and not a symlink? Just thinking about potential edge cases

Reply:
_Fixed in latest commit_

Reply:
nit: it is a bit harder to review changes when force pushing since the PR diff expands the entire file each time. Is there a better way for me to see changes as they come in?

Reply:
If you have ideas, I'm happy to improve my PR etiquette. Conversely if I had made a new commit for every review comment, I'd imagine it would be much harder to digest the history, and much harder for me to keep up-to-date.